### PR TITLE
Add an `Iterator.prototype.find` to the adventure pack

### LINKED ISSUE #79: Add `Iterator.prototype.find` to the adventure pack
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/find

### PR DESCRIPTION
**Pull request by elimanzo**
[find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/find) goody!

Closes #79.

### CODE REVIEW

File: tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            key === key.toUpperCase() &&
+            value.type === "fruit" &&
+            value.price < 3.0,
+        ),
+    ).toEqual(["GRAPES", { type: "fruit", price: 2.99 }]);
+  });
+
+  it("can test Set values()", () => {
+    const set = new Set([2, 2, -3, -3, 4, -5, 5, -6]);
+    expect(set.values().find((element) => element > 4)).toBe(5);
```
Comment (miorel) [inline tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:45]: To make the test more interesting, let's add another element that satisfies the requirement. Then we are also testing that `.find` finds the first one that matches!

File: tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    expect(array.values().find((element) => element > 0)).toBe(5);
+  });
+
+  it("returns undefined if no element satisfies the callback", () => {
+    const array = [-8, -3, -9, -2];
+    expect(array.values().find((element) => element > 0)).toBeUndefined();
+  });
+
+  it("returns undefined for an empty iterator", () => {
+    expect([].values().find(() => true)).toBeUndefined();
```
Comment (miorel) [inline tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:19]: Nice lambda!

File: tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    const generator = function* (): Generator<number, void, undefined> {
+      yield 2;
+      yield -4;
+      yield 6;
+      yield -8;
+      yield 3;
+    };
+    expect(generator().find((element) => element % 2 === 1)).toBe(3);
+  });
+});
```
Comment (miorel) [inline tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:58]: Similar to `.some` and `.every`, might be nice to test that we don't test the whole array.

File: tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
```
Comment (miorel) [inline tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:3]: The changes from #140 may be relevant here as well.

Comment (elimanzo) [reply tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:6]: ack!

File: tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts (hunk 1 above)
Comment (elimanzo) [reply tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:48]: ack!

File: tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts (hunk 3 above)
Comment (elimanzo) [reply tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:61]: ack!

Comment (miorel) [reply tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:61]: Will you be adding a test for this? I was thinking something that checks the invocation count using `jest.fn`.

Comment (elimanzo) [reply tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts:61]: oh I understand, yeah I'll write one up!

**Pull request merged**